bob-api/api/de_onderwijzers_api.php

```diff
@@ -0,0 +1,195 @@
+     *
+     * Create a new session token that will be used for all subsequent requests
+     * @throws ErrorException
+     */
+    public function __construct()
+    {
+        $this->applicationSessionToken = $this->generateSessionToken();
+    }
+
+    /**
+     * Get information about the currently logged in user
+     * @param string $sessionToken Session token for a user
+     * @return array
+     * @throws ErrorException
+     */
+    public function getMe(string $sessionToken): array
+    {
+        $this->logDebug("Executing getMe() query");
+
+        $query = <<<GRAPHQL
+        query RootQuery(\$sessionToken: UUID) {
+            viewer(sessionToken: \$sessionToken) {
+                me {
+                  id,
+                  email,
+                  apiToken,
+                  avatar
+                }
+            }
+        }
+GRAPHQL;
+
+        $result = $this->executeQuery($query, array("sessionToken" => $sessionToken));
+
+        $this->logDebug("Execution of getMe() finished");
+
+        return $result['data'];
+    }
+
+    /**
+     * Validate the given credentials
+     * @param string $email The email of the user
+     * @param string $password The password of the user
+     * @return array
+     * @throws ErrorException
+     */
+    public function validateCredentials(string $email, string $password): array
+    {
+        $this->logDebug("Executing validateCredentials() query");
+
+        $query = <<<GRAPHQL
+        query RootQuery(\$sessionToken: UUID, \$email: String, \$passwordHash: String, \$hardware: JSONString) {
+           viewer(sessionToken: \$sessionToken) {
+            validateCredentials(email: \$email, passwordHash: \$passwordHash, hardware: \$hardware) {
+              ok,
+              feedback,
+              user {
+                id,
+                email,
+                name,
+                permalink,
+                avatar
+              },
+              sessionToken
+            }
+          }
+        }
+GRAPHQL;
+
+        $passwordHash = md5($password);
+
+        $result = $this->executeQuery($query, array(
+            "sessionToken" => $this->applicationSessionToken,
+            "email" => $email,
+            "passwordHash" => $passwordHash,
+            "hardware" => "{\"test\": 1}"
+        ));
+
+        $this->logDebug("Execution of validateCredentials() finished");
+
+        return $result['data']['viewer']['validateCredentials'];
+    }
+
+    /**
+     * Generate a new session token
+     * @throws ErrorException
+     */
+    private function generateSessionToken(): string
+    {
+        $this->logDebug("Executing generateSessionToken() query");
+
+        $query = <<<GRAPHQL
+        query RootQuery(\$hardware: JSONString!, \$platform: Platform!) {
+            newSessionToken(hardware: \$hardware, platform: \$platform)
+        }
+GRAPHQL;
+
+        $result = $this->executeQuery($query, array("platform" => "DEBUG", "hardware" => "{\"test\": 1}"));
```

Comment:
ik vindt dat het platform naam 'DEBUG' een variable kan zijn, omdat we meerde platformen hebben gekregen: 'LERAAR' + 'DEBUG_LERAAR

Reply:
helemaal mee eens

Reply:
Prima voorstel. Misschien beter om de volgende keer: 'ik denk' in plaats van 'ik vind' te gebruiken, komt iets minder dwingend over :)